acme/provider.go

```diff
@@ -1,8 +1,24 @@
 package acme
 
+import "time"
+
 // ChallengeProvider presents the solution to a challenge available to be solved
 // CleanUp will be called by the challenge if Present ends in a non-error state.
 type ChallengeProvider interface {
 	Present(domain, token, keyAuth string) error
 	CleanUp(domain, token, keyAuth string) error
 }
+
+// ChallengeProviderTimeout allows for implementing a
+// ChallengeProvider where an unusually long timeout is required when
+// waiting for an ACME challenge to be satisfied. It can be used to
+// implement a DNS ChallengeProvider which requires a long timeout
+// when checking for DNS record progagation. If an implementor of a
+// ChallengeProvider also provides a Timeout method, then the return
```

Comment:
IMO the first two sentences here should be merged as they are basically saying the same thing.